index.js

```diff
@@ -107,40 +108,30 @@ class KeyringController extends EventEmitter {
    *
    * @fires KeyringController#unlock
    * @param {string} password - The password to encrypt the vault with.
-   * @param {string|Array<number>} seedPhrase - The BIP39-compliant seed phrase,
-   * either as a string or an array of UTF-8 bytes that represent the string.
+   * @param {Uint8Array | string} seedPhrase - The BIP39-compliant seed phrase,
+   * either as a string or Uint8Array.
    * @returns {Promise<object>} A Promise that resolves to the state.
    */
   async createNewVaultAndRestore(password, seedPhrase) {
-    const seedPhraseAsBuffer =
-      typeof seedPhrase === 'string'
-        ? Buffer.from(seedPhrase, 'utf8')
-        : Buffer.from(seedPhrase);
+    const seedPhraseUint8Array = this._mnemonicToUint8Array(seedPhrase);
 
     if (typeof password !== 'string') {
       throw new Error('Password must be text.');
     }
 
-    const wordlists = Object.values(bip39.wordlists);
-    if (
-      wordlists.every(
-        (wordlist) => !bip39.validateMnemonic(seedPhraseAsBuffer, wordlist),
-      )
-    ) {
-      throw new Error('Seed phrase is invalid.');
+    if (!bip39.validateMnemonic(seedPhraseUint8Array, wordlist)) {
+      throw new Error('KeyringController - Seed phrase is invalid.');
     }
 
     this.password = password;
 
-    this.clearKeyrings();
-    const firstKeyring = await this.addNewKeyring(
-      KEYRINGS_TYPE_MAP.HD_KEYRING,
-      {
-        mnemonic: seedPhraseAsBuffer,
-        numberOfAccounts: 1,
-      },
-    );
-    const [firstAccount] = await firstKeyring.getAccounts();
+    await this.clearKeyrings();
```

Comment:
Interesting; another `async` call without an `await`. In this case the function wasn't doing anything async, so this should result in no functional change.